### PR TITLE
feat(app-platform): UI Integrations Schema

### PR DESCRIPTION
**Pull request by mnoble**
Introduces the specification for UI Integration components. It's written in and uses JSON Schema to validate. When developers submit their UI Integration schema, this will be used to validate everything.

There are three categories of components that are usable:

## Form Field Components
Used as children of certain **Feature Components** that allow you to specify form inputs for Users to fill out.

## Composable Components
Used as children `elements` to create content in components like **Issue Media**.

## Feature Components
The top-level components you specify that enable specific UI features for Users.

This PR is just the schema definition and tests for each individual component. The work to use this specification to validate received schema will come later.

### CODE REVIEW

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            'type': 'string',
+            'format': 'uri',
+        },
+
+        'options': {
+            'type': 'array',
+            'items': {
+                'type': 'array',
+                'minItems': 2,
+                'maxItems': 2,
```
Comment (markstory) [inline src/sentry/api/validators/sentry_apps/schema.py]: Why only 2 items?

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+                'type': {
+                    'type': 'string',
+                    'enum': ['text'],
+                },
+                'label': {
+                    'type': 'string',
+                },
+                'name': {
+                    'type': 'string',
+                },
```
Comment (markstory) [inline src/sentry/api/validators/sentry_apps/schema.py]: Can fields be marked as required through an attribute?

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+                'url': {
+                    'type': 'string',
+                    'format': 'uri',
+                    'pattern': '^https?://',
+                },
+            },
+            'required': ['type', 'url'],
+        },
+
+        'markdown': {
```
Comment (markstory) [inline src/sentry/api/validators/sentry_apps/schema.py]: Markdown generally allows arbitrary HTML to be used. Be careful.

File: src/sentry/api/validators/sentry_apps/schema.py (hunk 1 above)
Comment (mnoble) [reply src/sentry/api/validators/sentry_apps/schema.py]: `[<label>, <value>]`

File: src/sentry/api/validators/sentry_apps/schema.py (hunk 3 above)
Comment (mnoble) [reply src/sentry/api/validators/sentry_apps/schema.py]: Yeaaaaaaa, we were talking about this this morning. This whole thing might need to be in an `iframe` as well.

File: src/sentry/api/validators/sentry_apps/schema.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/validators/sentry_apps/schema.py]: Makes sense :+1:

File: src/sentry/api/validators/sentry_apps/schema.py (hunk 3 above)
Comment (markstory) [reply src/sentry/api/validators/sentry_apps/schema.py]: You could also run something like [bleach](https://pypi.org/project/bleach/) over the rendered HTML allowing you to safely subset both markdown and HTML.

File: tests/sentry/api/validators/sentry_apps/test_select.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    def test_valid_schema_with_options(self):
+        validate_component(self.schema)
+
+    def test_valid_schema_options_with_numeric_value(self):
+        self.schema['options'][0][1] = 1
+        self.schema['options'][1][1] = 2
+
+        validate_component(self.schema)
+
+    def test_valid_schema_with_uri(self):
```
Comment (MeredithAnya) [inline tests/sentry/api/validators/sentry_apps/test_select.py]: would we also want to test an invalid uri format?

**Review by MeredithAnya — APPROVED** [review]

File: src/sentry/api/validators/sentry_apps/schema.py (hunk 2 above)
Comment (mnoble) [reply src/sentry/api/validators/sentry_apps/schema.py]: Doesn't seem like it 😒 

https://json-schema.org/understanding-json-schema/reference/object.html#required

File: tests/sentry/api/validators/sentry_apps/test_select.py (hunk 4 above)
Comment (mnoble) [reply tests/sentry/api/validators/sentry_apps/test_select.py]: We might want to look into using a regex to validate them in the future – I'm not sure the `format: uri` really does a whole lot currently. I think it's to flesh that out later.

**Pull request merged**